app/src/main/java/com/example/fitappa/Routine/ViewRoutineActivity.java

```diff
@@ -79,9 +80,11 @@ public void goBackToViewRoutines() {
 
     /**
      * This method initializes the RoutineView with the user's workouts.
+     *
+     * @param workoutTemplates represents the user's workouts
      */
-    private void initializeRoutineView() {
-        for (Workout w : routine) {
+    private void initializeRoutineView(List<WorkoutTemplate> workoutTemplates) {
```

Comment:
parameter in this method not needed. Change workoutTemplates from this method to routine (since routine is iterable we dont need to do routine.getWorkoutTemplates() anymore